pkg/store/cache/caching_bucket.go

```diff
@@ -26,92 +29,178 @@ import (
+		cache:      cache,
+		opConfig:   opConfig,
+	})
+
+	cb.operationRequests.WithLabelValues(operationName, configName)
+	cb.operationHits.WithLabelValues(operationName, configName)
+}
+
+// CacheIter configures caching of "Iter" operation for matching directories.
+func (cb *CachingBucket) CacheIter(configName string, cache cache.Cache, matcher func(string) bool, ttl time.Duration) {
+	cb.addOperationConfig(operationIter, configName, matcher, cache, iterConfig(ttl))
+}
+
+// CacheExists configures caching of "Exists" operation for matching files. Negative values are cached as well.
+func (cb *CachingBucket) CacheExists(configName string, cache cache.Cache, matcher func(string) bool, existsTTL, doesntExistTTL time.Duration) {
+	cb.addOperationConfig(operationExists, configName, matcher, cache, existsConfig{existsTTL, doesntExistTTL})
+}
+
+// CacheGet configures caching of "Get" operation for matching files. Content of the object is cached, as well as whether object exists or not.
+func (cb *CachingBucket) CacheGet(configName string, cache cache.Cache, matcher func(string) bool, contentTTL, existsTTL, doesntExistTTL time.Duration) {
+	cb.addOperationConfig(operationGet, configName, matcher, cache, getConfig{contentTTL: contentTTL, existsTTL: existsTTL, doesntExistTTL: doesntExistTTL})
+}
+
+// CacheGetRange configures caching of "GetRange" operation. Subranges (aligned on subrange size) are cached individually.
+// Since caching operation needs to know the object size to compute correct subranges, object size is cached as well.
+// Single "GetRange" requests can result in multiple smaller GetRange sub-requests issued on the underlying bucket.
+// MaxSubRequests specifies how many such subrequests may be issued. Values <= 0 mean there is no limit (requests
+// for adjacent missing subranges are still merged).
+func (cb *CachingBucket) CacheGetRange(configName string, cache cache.Cache, matcher func(string) bool, subrangeSize int64, objectSizeTTL, subrangeTTL time.Duration, maxSubRequests int) {
+	cb.addOperationConfig(operationGetRange, configName, matcher, cache, getRangeConfig{
+		subrangeSize:   subrangeSize,
+		objectSizeTTL:  objectSizeTTL,
+		subrangeTTL:    subrangeTTL,
+		maxSubRequests: maxSubRequests,
+	})
+
+	cb.requestedGetRangeBytes.WithLabelValues(configName)
+	cb.fetchedGetRangeBytes.WithLabelValues(originCache, configName)
+	cb.fetchedGetRangeBytes.WithLabelValues(originBucket, configName)
+	cb.refetchedGetRangeBytes.WithLabelValues(originCache, configName)
+}
+
+// CacheObjectSize configures caching of "ObjectSize" operation for matching files.
+func (cb *CachingBucket) CacheObjectSize(configName string, cache cache.Cache, matcher func(name string) bool, ttl time.Duration) {
+	cb.addOperationConfig(operationObjectSize, configName, matcher, cache, objectSizeCfg(ttl))
+}
+
+func (cb *CachingBucket) findCacheConfig(configs []*operationConfig, objectName string) (configName string, cache cache.Cache, cfg interface{}) {
```

Comment:
I would consider this method quite shallow. It's not adding much, let's just inline, or better .. replace configs with map?

Reply:
It is used from multiple places. In fact, I've now added more methods like this, to avoid using generic map for all kinds of configs.